web_monitoring/diffing_server.py

```diff
@@ -311,17 +311,17 @@ def check_response_for_error(self, response):
             try:
                 response.rethrow()
             except (ValueError, OSError, tornado.httpclient.HTTPError):
-                # Response code == 599 means that
-                # no HTTP response was received.
-                # In this case the error code should
-                # become 400 indicating that the error was
-                # raised because of a bad request parameter.
-                if response.code == 599:
+                url = response.request.url
+                # Timeouts aren't distinguishable from other network failures
+                # by status code, so check the actual error type.
+                if isinstance(response.error, tornado.simple_httpclient.HTTPTimeoutError):
                     self.send_error(
-                        400, reason=str(response.error))
+                        504,
+                        reason=f'Timed out while fetching "{url}"')
                 else:
                     self.send_error(
-                        response.code, reason=str(response.error))
+                        502,
+                        reason=f'Received a {response.code} status while fetching "{url}": {response.error}')
```

Comment:
Important question: if there was a network failure that wasn't a timeout, are we worried about this leaking sensitive information? You might get an error like:

```json
{
  "code": 502,
  "error": "Received a 599 status while fetching \"http://reallynowherethisdoesntexistpleasefail.org/web/20171227204051/https://el.wikipedia.org/wiki/\u0386\u03c1\u03b7\u03c2_(\u03c0\u03bb\u03b1\u03bd\u03ae\u03c4\u03b7\u03c2)\": [Errno 8] nodename nor servname provided, or not known"
}
```

I *think* it’s ok, but we could also have a custom clause for 599 status codes (network failures).